mcs/class/System/System.Net.NetworkInformation/GatewayIPAddressInformationCollection.cs

```diff
@@ -1,156 +0,0 @@
-//
-// Permission is hereby granted, free of charge, to any person obtaining
-// a copy of this software and associated documentation files (the
-// "Software"), to deal in the Software without restriction, including
-// without limitation the rights to use, copy, modify, merge, publish,
-// distribute, sublicense, and/or sell copies of the Software, and to
-// permit persons to whom the Software is furnished to do so, subject to
-// the following conditions:
-// 
-// The above copyright notice and this permission notice shall be
-// included in all copies or substantial portions of the Software.
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-//
-using System.Collections;
-using System.Collections.Generic;
-using System.Runtime.InteropServices;
-
-namespace System.Net.NetworkInformation {
-	public class GatewayIPAddressInformationCollection : ICollection<GatewayIPAddressInformation>, IEnumerable<GatewayIPAddressInformation>, IEnumerable {
-		List<GatewayIPAddressInformation> list = new List<GatewayIPAddressInformation> ();
-		
-		protected GatewayIPAddressInformationCollection ()
-		{
-		}
-
-		public virtual void Add (GatewayIPAddressInformation address)
-		{
-			if (IsReadOnly)
-				throw new NotSupportedException ("The collection is read-only.");
-			list.Add (address);
-		}
-
-		public virtual void Clear ()
-		{
-			if (IsReadOnly)
-				throw new NotSupportedException ("The collection is read-only.");
-			list.Clear ();
-		}
-
-		public virtual bool Contains (GatewayIPAddressInformation address)
-		{
-			return list.Contains (address);
-		}
-
-		public virtual void CopyTo (GatewayIPAddressInformation [] array, int offset)
-		{
-			list.CopyTo (array, offset);
-		}
-
-		public virtual IEnumerator<GatewayIPAddressInformation> GetEnumerator ()
-		{
-			return ((IEnumerable<GatewayIPAddressInformation>)list).GetEnumerator ();
-		}
-
-		public virtual bool Remove (GatewayIPAddressInformation address)
-		{
-			if (IsReadOnly)
-				throw new NotSupportedException ("The collection is read-only.");
-			return list.Remove (address);
-		}
-
-		IEnumerator IEnumerable.GetEnumerator ()
-		{
-			return list.GetEnumerator ();
-		}
-
-		public virtual int Count {
-			get { return list.Count; }
-		}
-
-		public virtual bool IsReadOnly {
-			get { return true; }
-		}
-
-		public virtual GatewayIPAddressInformation this [int index] {
-			get { return list [index]; }
-		}
-	}
-
-#if !MOBILE
-	class Win32GatewayIPAddressInformationCollection : GatewayIPAddressInformationCollection
```

Comment:
This class was excluded from the mobile build in https://github.com/mono/mono/commit/9ea2f50a2adefdeadefbf8f49aa94404e3175a1f, but in the new code it isn't. Do the reasons from that commit still apply?